File: src/main/kotlin/ru/ifmo/ctddev/segal/hw2/IterativeStopPredicate.kt
package ru.ifmo.ctddev.segal.hw2

/**
 * Created by dimatomp on 27.03.16.
 */
sealed class IterativeStopPredicate {
    class NumIterations(val nIter: Int): IterativeStopPredicate()
    class EpsilonStop(val eps: Double): IterativeStopPredicate()
}

File: src/main/scala/ru/ifmo/ctddev/segal/hw2/IterativeStopPredicate.scala
package ru.ifmo.ctddev.segal.hw2

/**
  * @author itegulov
  */
sealed trait IterativeStopPredicate

case class NumIterationsPredicate(iterations: Int) extends IterativeStopPredicate

case class EpsPredicate(eps: Double) extends IterativeStopPredicate
